libs-scala/grpc-reverse-proxy/src/main/scala/com/daml/grpc/ReverseProxy.scala

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.grpc
+
+import com.daml.grpc.reflection.ServerReflectionClient
+import io.grpc._
+import io.grpc.reflection.v1alpha.ServerReflectionGrpc
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ReverseProxy {
+
+  def create(
+      backend: Channel,
+      serverBuilder: ServerBuilder[_],
+  )(implicit
+      ec: ExecutionContext
+  ): Future[Server] = create(backend, serverBuilder, Map.empty[String, Seq[ServerInterceptor]])
+
+  def create(
+      backend: Channel,
+      serverBuilder: ServerBuilder[_],
+      interceptor: (String, Seq[ServerInterceptor]),
```

Comment:
Why do we special-case the first one here?

Reply:
This "pattern" forces the called to provide at least one interceptor.

Reply:
it does but then you have another overload that works without supplying an interceptor which seems to make that redundant?

Reply:
Good point, I'll remove it.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/8481/commits/b641b9e8e409240cdfb83f701a036de09d8a2b97